### PR TITLE
Random: cut randstring allocation size in half

### PR DESCRIPTION
**Pull request by vtjnash**
```
julia> @time randstring(300) # master
  0.000004 seconds (2 allocations: 736 bytes)

julia> @time randstring(300) # PR
  0.000003 seconds (2 allocations: 400 bytes)
```
voila

### CODE REVIEW

File: stdlib/Random/src/misc.jl [hunk 1]
```diff
@@ -71,7 +71,7 @@ function randstring end
 
 let b = UInt8['0':'9';'A':'Z';'a':'z']
     global randstring
-    randstring(r::AbstractRNG, chars=b, n::Integer=8) = String(rand(r, chars, n))
+    randstring(r::AbstractRNG, chars=b, n::Integer=8) = String(rand!(r, Base.StringVector(n), chars))
```
Comment (rfourquet) [inline stdlib/Random/src/misc.jl:74]: It doesn't seem to work in the général case, e.g. with this change:
```julia
julia> randstring("àé", 2)
"\xe0\xe9"
```

Comment (rfourquet) [reply stdlib/Random/src/misc.jl:74]: But this could be made to work in the case where `chars::Vector{UInt8}` at least, or when `isascii(chars)`.

**Comment by rfourquet** [comment]
Would you mind clarifying why `String(Base.StringVector(300))` allocates less than `String(Vector{UInt8}(undef, 300))`, given that the argument to `String` seems to produce exactly the same thing?

**Review by rfourquet — APPROVED** [review]

**Review by StefanKarpinski — APPROVED** [review]

File: stdlib/Random/src/misc.jl (hunk 1 above)
Comment (vtjnash) [reply stdlib/Random/src/misc.jl]: The `isascii` test is an interesting thought. That would probably mostly even avoid the performance gotcha mentioned in the docstring.

**Pull request merged**